### PR TITLE
KEYCLOAK-7310 Add migration test from 3.4.x to 4.x

### PR DESCRIPTION
**Pull request by vramik**

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/MigrationTest.java [hunk 1]
```diff
@@ -58,18 +59,28 @@ public void beforeMigrationTest() {
...
+        testMigratedData();
+        testMigrationTo4_x();
+    }
+
     @Test
     @Migration(versionFrom = "2.5.5.Final")
     public void migration2_5_5Test() {
         testMigratedData();
-        testMigrationTo3_x_and_higher();
+        testMigrationTo3_x();
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/MigrationTest.java]: It seems that until now, we tested migration from 2.5.5 to all the versions (including newest "testMigrationTo4_x"). With this change, we will test from 2.5.5 just to 3.x, but not to latest 4.x if I understand correctly. Is this the change we want to introduce? Shouldn't we always test to newest version?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/MigrationTest.java [hunk 2]
```diff
@@ -80,7 +91,7 @@ public void migration1_9_8Test() throws Exception {
         testMigrationTo2_3_0();
         testMigrationTo2_5_0();
         testMigrationTo2_5_1();
-        testMigrationTo3_x_and_higher();
+        testMigrationTo3_x();
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/MigrationTest.java]: Same comment as above

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport198MigrationTest.java [hunk 3]
```diff
@@ -70,9 +70,10 @@ public void migration1_9_8Test() throws Exception {
         testMigrationTo2_3_0();
         testMigrationTo2_5_0();
         //testMigrationTo2_5_1(); // Offline tokens migration is skipped for JSON
-        testMigrationTo3_x_and_higher();
+        testMigrationTo3_x();
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport198MigrationTest.java]: Same comment as below

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport255MigrationTest.java [hunk 4]
```diff
@@ -64,7 +64,7 @@ public void addTestRealms(List<RealmRepresentation> testRealms) {
 
     @Test
     public void migration2_5_5Test() throws Exception {
-        testMigrationTo3_x_and_higher();
+        testMigrationTo3_x();
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/migration/JsonFileImport255MigrationTest.java]: Same comment as below

**Review by mposolda — COMMENTED** [review]
Added few comments/questions

**Comment by vramik** [comment]
@mposolda thanks. I didn't want to introduce the change. I wanted to unify the tests. It should be fixed now

**Review by mposolda — APPROVED** [review]
@vramik Cool, Thanks!

**Pull request merged**